acceptance/tests/provider/package/puppetserver_gem.rb

```diff
@@ -0,0 +1,41 @@
+test_name "puppetserver_gem provider should install and uninstall" do
+  tag 'audit:high',
+      'server'
+
+  require 'puppet/acceptance/common_utils'
+  extend Puppet::Acceptance::PackageUtils
+  extend Puppet::Acceptance::ManifestUtils
+
+  skip_test 'puppetserver_gem is only suitable on server nodes' unless master
+
+  package = 'world_airports'
+
+  agents.each do |agent|
+    if agent['roles'].include?('master')
+      teardown do
+        # Ensure the gem is uninstalled if anything goes wrong
+        # TODO maybe execute this only if something fails, as it takes time
+        on(agent, "puppetserver gem uninstall #{package}")
+      end
```

Comment:
Maybe we can use the `master` keyword to handle those things?
Also i think this the teardown can be registered only once, i don't believe there are multiple masters environments.
```ruby
teardown do
  # Ensure the gem is uninstalled if anything goes wrong
  # TODO maybe execute this only if something fails, as it takes time
  on(master, "puppetserver gem uninstall #{package}")
end
```

Reply:
In theory it should be the same, as we're only selecting server agents on line 14, but I can change it.

Reply:
Ah I see what you mean, we can drop the whole `agents.each` block.